web/src/__tests__/CityContentSwitcher.spec.tsx

```diff
@@ -28,43 +29,58 @@ describe('CityContentSwitcher', () => {
   const [city, cityWithDisabledFeatures] = new CityModelBuilder(2).build() as [CityModel, CityModel]
   const path = (city: CityModel, routeName: string) => `/${city.code}/${languageCode}/${routeName}`
 
+  const MockComponent = () => {
+    const pathname = normalizePath(useLocation().pathname)
+    return <div>{pathname}</div>
+  }
+  const renderCityContentSwitcher = (routeName: string) =>
+    renderRoute(
+      <>
+        <CityContentSwitcher languageCode={languageCode} />
+        <MockComponent />
+      </>,
+      {
+        routePattern: cityContentPattern,
+        pathname: path(city, routeName),
+      },
+    )
+
   it.each([{ routeName: SEARCH_ROUTE }, { routeName: DISCLAIMER_ROUTE }, { routeName: POIS_ROUTE }])(
     'should navigate to $routeName route',
     async ({ routeName }) => {
       mockUseLoadFromEndpointWithData(city)
-      const { findByText } = renderRoute(<CityContentSwitcher languageCode={languageCode} />, {
-        routePattern: cityContentPattern,
-        pathname: path(city, routeName),
-      })
+      const { findByText } = renderCityContentSwitcher(routeName)
```

Comment:
Any chance of using a `findByRole` here and `getByRole` in the later tests?

Reply:
Since the way roles are handled will change in the react-native upgrade I'd rather not :)